core/src/test/scala/org/apache/spark/network/netty/NettyBlockTransferServiceSuite.scala

```diff
@@ -77,16 +79,53 @@ class NettyBlockTransferServiceSuite
     verifyServicePort(expectedPort = service0.port + 1, actualPort = service1.port)
   }
 
+  test("can bind to two max specific ports") {
+    service0 = createService(port = 65535)
+    service1 = createService(port = 65535)
+    verifyServicePort(expectedPort = 65535, actualPort = service0.port)
+    // see `Utils.userPort` the user port to try when trying to bind a service,
+    // the max privileged port is 1024.
+    verifyServicePort(expectedPort = 1024, actualPort = service1.port)
+  }
+
+  test("can't bind to a privileged port") {
+    intercept[IllegalArgumentException] {
+      service0 = createService(port = 23)
+    }
+  }
+
+  test("turn off spark.port.maxRetries, bind repeat port is fail") {
+    val port = 17634 + Random.nextInt(10000)
+    logInfo("random port for test: " + port)
+    service0 = createService(port)
+
+    // `service0.port` is occupied, bind repeat port throw BindException.
+    intercept[BindException] {
```

Comment:
I still don't think this is something to test. You're basically testing Netty's API